.github/workflows/release.yml

```diff
@@ -349,6 +349,47 @@ jobs:
             const { owner, repo } = context.repo
             await github.rest.repos.deleteRelease({ owner, repo, release_id: ${{ needs.set-release-id.outputs.release_id }} })
 
+  pyk-publish:
+    name: 'Publish pyk'
+    runs-on: [self-hosted, linux, flyweight]
+    environment: production
+    permissions:
+      id-token: write
+    needs: [set-release-id, source-tarball]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: '3.10'
+
+      - name: Install Poetry
+        uses: Gr1N/setup-poetry@v9
+        with:
+          python-version: '3.10'
+    
+      - name: Setup Caching
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/poetry/virtualenvs
+          key: ${{ runner.os }}-poetry-${{ hashFiles('pyk/pyproject.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-poetry-
```

Comment:
There's not much to cache, `poetry build` does not require a virtual environment.

```suggestion
```